lib/auth/oauth/oauth.ts

```diff
@@ -39,6 +40,25 @@ export interface CallbackResponse<T = AdapterHeaders> {
   session: Session;
 }
 
+interface BotCheckArgs {
+  request: AdapterRequest;
+  shop: string;
+  log: ShopifyLogger;
+}
+
+const respondToPossibleBotRequest = (args: BotCheckArgs) => {
+  const {request, shop, log} = args;
+  if (request.headers['User-Agent']?.includes('GoogleOther') || !shop) {
```

Comment:
I wonder if checking the shop here is the right thing to do. We have an explicit check later on which will throw a more specific error message.

Is the concern here that there might be other bots that will follow the same pattern? In that case, would it make more sense to check that we have **no** query args?

Reply:
I like this idea. I'll change that